nilearn/_utils/niimg_conversions.py

```diff
@@ -237,66 +286,45 @@ def check_niimgs(niimgs, accept_3d=False, return_iterator=False):
 
     Notes
     -----
-    This function is the equivalent to check_niimg() for Niimg-like objects
+    This function is the equivalent to check_niimg_3d() for Niimg-like objects
     with a session level.
 
     Its application is idempotent.
     """
-    # Initialization:
-    # If given data is a list, we count the number of levels to check
-    # dimensionality and make a consistent error message.
-    depth = 0
-    first_img = niimgs
-    if accept_3d and (isinstance(first_img, string_types)
-                      or not isinstance(first_img, collections.Iterable)):
-        niimg = check_niimg(niimgs)
-        if len(niimg.shape) == 3:
-            niimg = new_img_like(niimg, niimg.get_data()[..., np.newaxis],
-                            niimg.get_affine())
-        return niimg
+    # Two types of input:
+    # * 4D nifti image
+    # * list of 3D niimgs
```

Comment:
I didn't put it here but iterator should work if you do `return_iterator=True` (not the other case as `concat_niimgs` is not iterator friendly). But, again, the goal of this PR is not to change the behavior of the code when using iterators. It is exactly as broken as before ;)